framework/Optimizers/GradientDescent.py

```diff
@@ -0,0 +1,840 @@
+    if len(stepNode.subparts) != 1:
+      self.raiseAnError('The <stepNode> node requires exactly one stepping strategy! Choose from: ', stepKnownTypes())
+    stepNode = next(iter(stepNode.subparts))
+    stepType = stepNode.getName()
+    self._stepInstance = stepReturnInstance(stepType, self)
+    self._stepInstance.handleInput(stepNode)
+
+    # acceptance strategy
+    acceptNode = paramInput.findFirst('acceptance')
+    if acceptNode:
+      if len(acceptNode.subparts) != 1:
+        self.raiseAnError('The <acceptance> node requires exactly one acceptance strategy! Choose from: ', acceptKnownTypes())
+      acceptNode = next(iter(acceptNode.subparts))
+      acceptType = acceptNode.getName()
+      self._acceptInstance = acceptReturnInstance(acceptType, self)
+      self._acceptInstance.handleInput(acceptNode)
+    else:
+      # default to strict mode acceptance
+      acceptNode = acceptReturnInstance('Strict', self)
+
+    # convergence options
+    convNode = paramInput.findFirst('convergence')
+    if convNode is not None:
+      for sub in convNode.subparts:
+        if sub.getName() == 'persistence':
+          self._requiredPersistence = sub.value
+        elif sub.getName() == 'terminateFollowers':
+          self._terminateFollowers = sub.value
+          self._followerProximity = sub.parameterValues.get('proximity', 1e-2)
+        else:
+          self._convergenceCriteria[sub.name] = sub.value
+    if not self._convergenceCriteria:
+      self.raiseAWarning('No convergence criteria given; using defaults.')
+      self._convergenceCriteria['gradient'] = 1e-6
+    # same point is ALWAYS a criterion
+    self._convergenceCriteria['samePoint'] = 1e-16 # TODO user option?
+    # set persistence to 1 if not set
+    if self._requiredPersistence is None:
+      self.raiseADebug('No persistence given; setting to 1.')
+      self._requiredPersistence = 1
+
+  def initialize(self, externalSeeding=None, solutionExport=None):
+    """
+      This function should be called every time a clean optimizer is needed. Called before takeAstep in <Step>
+      @ In, externalSeeding, int, optional, external seed
+      @ In, solutionExport, DataObject, optional, a PointSet to hold the solution
+      @ Out, None
+    """
+    RavenSampled.initialize(self, externalSeeding=externalSeeding, solutionExport=solutionExport)
+    self._gradientInstance.initialize(self.toBeSampled, self._gradProximity)
+    self._stepInstance.initialize(self.toBeSampled, persistence=self._requiredPersistence)
+    self._acceptInstance.initialize()
+    # if single trajectory, turn off follower termination
+    if len(self._initialValues) < 2:
+      self.raiseADebug('Setting terminateFollowers to False since only 1 trajectory exists.')
+      self._terminateFollowers = False
+    # queue up the first run for each trajectory
+    initialStepSize = self._stepInstance.initialStepSize(len(self.toBeSampled)) # TODO user scaling option
```

Comment:
we can add this in the future; I thought about adding it but technically if the user gives this value, we should add a bunch of sanity checks to assure this step size is reasonable and doesn't leave the problem domain, etc. It would be easy to add as soon as we have a use case for it.